### PR TITLE
Fix required indicator

### LINKED ISSUE #2601: Allow the indicator to be entered as text
### Description

Allow the indicator to be entered as text.

```tsx
  <FormControl
    isRequired
    label="Email address"
    requiredIndicator="（required）"
  >
    <Input type="email" placeholder="your email address" />
  </FormControl>
```

image

### Problem Statement/Justification

I want to be able to use the asterisk style here.

image

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by illionillion**
Closes #2601

### CODE REVIEW

File: packages/components/form-control/src/form-control.tsx [hunk 1]
```diff
@@ -425,8 +427,10 @@ export const RequiredIndicator = forwardRef<RequiredIndicatorProps, "span">(
         __css={css}
         {...rest}
       >
-        *
+        {children ? children : <>*</>}
```
Comment (hirotomoyamada) [inline packages/components/form-control/src/form-control.tsx:430]: Change `{children ? children : <>*</>}` to `{children ?? <>*</>}`

File: packages/components/form-control/src/form-control.tsx [hunk 2]
```diff
@@ -411,12 +413,12 @@ export const Label = forwardRef<LabelProps, "label">(
...
 
 export const RequiredIndicator = forwardRef<RequiredIndicatorProps, "span">(
-  ({ className, ...rest }, ref) => {
+  ({ className, children, ...rest }, ref) => {
     const styles = useFormControlStyles() ?? {}
 
     const css: CSSUIObject = { ...styles.requiredIndicator }
 
-    return (
+    return typeof children !== "object" ? (
```
Comment (hirotomoyamada) [inline packages/components/form-control/src/form-control.tsx:421]: Change `typeof children !== "object"` to `!isValidElement(children)` and add `import {  isValidElement } from "react"`.

**Comment by hirotomoyamada** [comment]
@illionillion 

Hi, thanks for the PR.

The `Fieldset` also uses the `requiredIndicator`. Do you want to change that in this PR as well?

File: packages/components/form-control/src/form-control.tsx (hunk 1 above)
Comment (illionillion) [reply packages/components/form-control/src/form-control.tsx]: I fixed!

File: packages/components/form-control/src/form-control.tsx (hunk 2 above)
Comment (illionillion) [reply packages/components/form-control/src/form-control.tsx]: I fixed.

**Comment by illionillion** [comment]
> @illionillion
> 
> Hi, thanks for the PR.
> 
> The `Fieldset` also uses the `requiredIndicator`. Do you want to change that in this PR as well?

I will change it.
Also add changeset.

**Comment by illionillion** [comment]
I fixed `fieldset`.
Please review again.

**Review by hirotomoyamada — APPROVED** [review]

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**